model/task/db.go

```diff
@@ -2962,3 +2962,46 @@ func getGenerateTasksEstimation(ctx context.Context, project, buildVariant, disp
 
 	return results, nil
 }
+
+// GetLatestTaskFromImage retrieves the latest task from all the distros corresponding to the imageID.
+func GetLatestTaskFromImage(ctx context.Context, imageID string) (*Task, error) {
+	distros, err := distro.GetDistrosForImage(ctx, imageID)
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving distros from imageID")
+	}
+	distroNames := make([]string, len(distros))
+	for i, d := range distros {
+		distroNames[i] = d.Id
+	}
+	if len(distroNames) == 0 {
+		return nil, errors.New("no distros found for image")
+	}
+	pipeline := []bson.M{
+		{
+			"$match": bson.M{
+				DistroIdKey: bson.M{
+					"$in": distroNames,
+				},
+			},
+		},
+		{
+			"$sort": bson.M{FinishTimeKey: -1},
+		},
+		{
+			"$limit": 1,
+		},
+	}
+	env := evergreen.GetEnvironment()
+	cursor, err := env.DB().Collection(Collection).Aggregate(ctx, pipeline)
+	if err != nil {
+		return nil, errors.Wrap(err, "finding latest task")
+	}
+	if cursor.Next(ctx) {
+		var task Task
```

Comment:
nit: I don't think there's something wrong with this but I think it's more idiomatic to use `task := Task{}` instead of var 

[Example](https://github.com/bynn/evergreen/blob/4b395be125cbb5cf404842a7287cc1ec0a598d98/model/version.go#L490)